src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,39 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cannot save movie to DB" + movie, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        Session session = null;
+        try {
```

Comment:
you can use try-with-resources  in this method 
<img width="715" alt="image" src="https://user-images.githubusercontent.com/96740815/180150555-80ddeec1-a483-4d89-b6ee-0367819b8983.png">